dynamiq/nodes/agents/base.py

```diff
@@ -644,13 +644,20 @@ def execute(
                 self._prompt_blocks["role"] = f"{{% raw %}}{self.role}{{% endraw %}}"
 
         files = input_data.files
+        uploaded_file_names: set[str] = set()
         if files:
             if not self.file_store_backend:
                 self.file_store = FileStoreConfig(enabled=True, backend=InMemoryFileStore())
                 self.tools.append(FileReadTool(file_store=self.file_store.backend, llm=self.llm))
                 self.tools.append(FileListTool(file_store=self.file_store.backend))
                 self._init_prompt_blocks()
-            self._ensure_named_files(files)
+            normalized_files = self._ensure_named_files(files)
```

Comment:
Please update retunr annotations in `def _ensure_named_files` as it returns list